wallet/chain/p/builder/builder.go

```diff
@@ -1068,31 +1146,61 @@ func (b *builder) spend(
 		})
 	}
 
-	// Iterate over the unlocked UTXOs
-	for _, utxo := range utxos {
+	// AVAX is handled last to account for fees.
+	avaxUTXOs, nonAVAXUTXOs := splitUTXOsByAssetID(unlockedUTXOs, b.context.AVAXAssetID)
```

Comment:
Is the logic here similar enough to the locked/unlocked case that it's worth abstracting the commonalities further?

Reply:
I had considered implementing a single `filter` function that took in a slice and a function... I felt like it was a bit overkill for this... But if you feel like that would be better... I could be convinced.